### PR TITLE
New HMI for single-button (user button) platforms

### PR DESCRIPTION
**Pull request by Peque**

### CODE REVIEW

File: hmi.c [hunk 1]
```diff
@@ -39,39 +39,69 @@ void blink_collision(void)
...
  *
- * Readings are performed each system clock tick.
+ * A user button response can either be an action or no-response.
  *
- * @param[in] count Required number of positive reads.
+ * @return The user button response.
  */
-bool button_right_read_consecutive(uint32_t count)
+enum button_response button_user_response(void)
 {
```
Comment (cua-cua) [inline hmi.c:66]: Macro for 500, 300 and 50 (equivalence in time maybe) :angel:

File: hmi.c [hunk 2]
```diff
@@ -39,39 +39,69 @@ void blink_collision(void)
...
 
 /**
- * @brief Read right button, requiring consecutive positive reads.
+ * @brief Check for any user button response.
  *
- * Readings are performed each system clock tick.
+ * A user button response can either be an action or no-response.
  *
- * @param[in] count Required number of positive reads.
+ * @return The user button response.
```
Comment (cua-cua) [inline hmi.c:63]: Include the possible response: SHORT, LONG and NONE

File: hmi.c [hunk 3]
```diff
@@ -94,25 +124,22 @@ void speaker_play_error(void)
 }
 
 /**
- * @brief Play three fast, high tones to note a successful operation.
+ * @brief Play a single and short high tone.
```
Comment (cua-cua) [inline hmi.c]: Include @param beeps

File: hmi.c [hunk 4]
```diff
@@ -94,25 +124,22 @@ void speaker_play_error(void)
 }
 
 /**
```
Comment (cua-cua) [inline hmi.c:123]: The number of short high tones is the input beeps of the function, it is not playing a single tone.

File: hmi.c [hunk 5]
```diff
@@ -181,46 +205,19 @@ float hmi_configure_force(float minimum_force, float force_step)
...
+		case BUTTON_LONG:
 			led_left_on();
 			led_right_on();
-			sleep_ticks(2000);
+			sleep_ticks(1000);
 			return force * force_step + minimum_force;
 		}
 	}
 }
-
```
Comment (cua-cua) [inline hmi.c:199]: Where is this performed?

File: hmi.c [hunk 6]
```diff
@@ -181,46 +205,19 @@ float hmi_configure_force(float minimum_force, float force_step)
 	uint8_t force = 0;
```
Comment (cua-cua) [inline hmi.c:207]: button_user_wait_action() already includes a while(1), remove this one.

**Review by cua-cua — CHANGES_REQUESTED** [review]

File: hmi.h [hunk 7]
```diff
@@ -10,17 +10,20 @@
...
 void speaker_play_success(void);
-void speaker_play_button(void);
 void speaker_play_competition(void);
-bool button_left_read_consecutive(uint32_t count);
-bool button_right_read_consecutive(uint32_t count);
+enum button_response button_user_response(void);
+enum button_action button_user_wait_action(void);
 void wait_front_sensor_close_signal(float close_distance);
-void initialize_solver_direction(void);
+void configure_solver_direction(void);
```
Comment (cua-cua) [inline hmi.h:24]: If reuse_maze is removed from .c, remove also from here. Delete also #include "mmlib/solve.h" dependency.

**Comment by cua-cua** [comment]
As you are already modifying HMI module.. it would be great to remove speed.h dependency from hmi.h as it is not being used anymore.

**Review by cua-cua — CHANGES_REQUESTED** [review]

File: hmi.c (hunk 2 above)
Comment (Peque) [reply hmi.c:63]: It is an enum! Doxygen (or whatever) should be able to retrieve that for us, should not it?

File: hmi.h (hunk 7 above)
Comment (Peque) [reply hmi.h:24]: Good catch! I guess [we could use some help](https://github.com/Bulebots/bulebule/issues/57) some day for these type of things. :joy:

bee650f363adb2cfeafc76e98e2ac16958c4813e

File: hmi.c (hunk 1 above)
Comment (Peque) [reply hmi.c:66]: Yeah... do not want to add 3 macros for that... But completely agree that it has to be implemented in seconds. I was lazy the first time, but it is actually cleaner this way:

42bd7dfeae037f5145ea8728973a2f1e3a7d5a87

File: hmi.c (hunk 3 above)
Comment (Peque) [reply hmi.c]: 272a76053b8ccd84acb4e8ec3ed05f7ff5930013

File: hmi.c (hunk 4 above)
Comment (Peque) [reply hmi.c:123]: 272a76053b8ccd84acb4e8ec3ed05f7ff5930013

Put the comment bellow the line! :joy:

File: hmi.c (hunk 5 above)
Comment (Peque) [reply hmi.c:199]: Comment bellow the line. :innocent: 

I guess you refer to the `reuse_maze()` function.

Now `main.c` takes care of that as part of the configuration process. We may move all that code from there to the HMI module, but I think we should not do that now.

File: hmi.c (hunk 6 above)
Comment (Peque) [reply hmi.c:207]: Comment bellow the line. :cry:

Actually, it is correct this way. We wait for user input (`button_user_wait_action()` loop), but then we are in another loop which allows us to increase the configured maximum force while running.

**Comment by Peque** [comment]
@cua-cua Removed `speed.h` dependency as suggested. :blush:

File: hmi.c (hunk 1 above)
Comment (cua-cua) [reply hmi.c:66]: Better like this :D

File: hmi.c (hunk 2 above)
Comment (cua-cua) [reply hmi.c:63]: mmm, okeeey

File: hmi.c (hunk 4 above)
Comment (cua-cua) [reply hmi.c:123]: puufff, I changed it, I never remember how to do that :sweat_smile:

File: hmi.c (hunk 5 above)
Comment (cua-cua) [reply hmi.c:199]: :+1:

File: hmi.h (hunk 7 above)
Comment (cua-cua) [reply hmi.h:24]: :)

File: hmi.c (hunk 6 above)
Comment (cua-cua) [reply hmi.c:207]: I don't like so many while(1)... but yep, now I understand your point.

**Review by cua-cua — APPROVED** [review]

**Pull request merged**